### PR TITLE
Esad/build block with gas used

### PR DESCRIPTION
**Pull request by eyusufatik**
# Description
Extending on @rakanalh 's work a new method to build blocks in the sequencer.

### CODE REVIEW

File: crates/sequencer/src/sequencer.rs [hunk 1]
```diff
@@ -695,8 +798,11 @@ where
 
     fn get_best_transactions(
         &self,
-        system_tx_gas_usage: u64,
-    ) -> anyhow::Result<Vec<RlpEvmTransaction>> {
+    ) -> anyhow::Result<
+        std::boxed::Box<
```
Comment (kpp) [inline crates/sequencer/src/sequencer.rs:802]:
```suggestion
        Box<
```

**Review by rakanalh — APPROVED** [review]

**Review by rakanalh — APPROVED** [review]

File: crates/evm/src/hooks.rs [hunk 2]
```diff
@@ -253,8 +253,6 @@ where
 
             tx_index += 1
         }
-
-        self.pending_transactions.clear(working_set);
```
[the remark this answers is not in the document]
Comment (rakanalh) [reply crates/evm/src/hooks.rs:257]: Probably introduced in different PRs but i saw it in this PR as redundant.

File: crates/sequencer/src/sequencer.rs [hunk 3]
```diff
@@ -282,25 +406,12 @@ where
...
-                let rlp_txs = match l2_block_mode {
-                    L2BlockMode::Empty => vec![],
-                    L2BlockMode::NotEmpty => self.get_best_transactions(system_tx_gas_usage)?,
-                };
-                debug!(
-                    "Sequencer: publishing block with {} transactions",
-                    rlp_txs.len()
-                );
-                let call_txs = CallMessage { txs: rlp_txs };
+                tracing::error!("we are out of tx selection!");
```
Comment (eyusufatik) [inline crates/sequencer/src/sequencer.rs:409]: let's remove this.

File: crates/sequencer/src/sequencer.rs [hunk 4]
```diff
@@ -214,6 +216,105 @@ where
...
+            &state_root,
+            prestate.clone(),
+            Default::default(),
+            &da_block_header,
+            &mut signed_batch,
+        ) {
+            (Ok(()), batch_workspace) => {
+                let block_gas_limit = self.db_provider.cfg().block_gas_limit;
+
+                let mut working_set_to_discard = batch_workspace.checkpoint().to_revertable();
```
Comment (eyusufatik) [inline crates/sequencer/src/sequencer.rs:243]: no need for it now

File: crates/sequencer/src/sequencer.rs [hunk 5]
```diff
@@ -214,6 +216,105 @@ where
...
+
+                let evm = Evm::<C>::default();
+
+                // while cumulative_gas_used <= 30_000_000 - 21000 {
```
Comment (eyusufatik) [inline crates/sequencer/src/sequencer.rs:247]: remove

**Pull request merged**